vaapi/VaapiBuffer.cpp

```diff
@@ -52,6 +52,27 @@ BufObjectPtr VaapiBuffer::create(const ContextPtr& context,
     return buf;
 }
 
+BufObjectPtr VaapiBuffer::create(const ContextPtr& context,
```

Comment:
full of duplicated code. Instead of pass num to this create function, could we just pass size*num to the original version of create? or at least original version of creat function can call this version of create like this:

create(context, type, size, data, 1)

Reply:
thanks @xuguangxin , I will modify the code and double-check its function.
but here, one more explaination for this piece of code:
I follow the code in [va-uilts](https://github.com/01org/libva-utils/blob/master/videoprocess/vavpp.cpp#L1273)
`va_status = vaCreateBuffer(va_dpy, context_id,
                               VAProcFilterParameterBufferType, sizeof(color_balance_param), **4**,
                               color_balance_param, &color_balance_param_buf_id);`
I am not sure if using **size*num** instead of this piece of code will be a potential bug.

Reply:
sorry about that @xuguangxin , you are right, I will modify the function like that:
    static BufObjectPtr **create**(const ContextPtr&,
        VABufferType type,
        uint32_t size,
        const void* data = 0,
        void** mapped = 0,
        **uint32_t num = 1);**

**size*num** doesn't work in this case, so add one format parameter **uint32_t num = 1**.